src/sourcekit-lsp/peekDocuments.ts

```diff
@@ -16,27 +16,82 @@ import * as vscode from "vscode";
 import * as langclient from "vscode-languageclient/node";
 import { PeekDocumentsParams, PeekDocumentsRequest } from "./lspExtensions";
 
+/**
+ * Opens a peeked editor in `uri` at `position` having contents from `locations`.
+ *
+ * **NOTE**:
+ * - If the `uri` is not open in the editor, this opens the `uri` in the editor and then opens a peeked editor.
+ * - This reopens a peeked editor in `uri` **if and only if** the previous peeked editor is being displayed at a
+ *   different position in `uri`.
+ * - This always closes any previously displayed peeked editor in `uri`.
+ *
+ * @param uri The uri of the file in which a peeked editor is to be opened
+ * @param position The position in the file in which a peeked editor is to be opened
+ * @param locations The locations of the contents which has to be displayed by the peeked editor
+ */
+async function openPeekedEditorIn(
+    uri: vscode.Uri,
+    position: vscode.Position,
+    locations: vscode.Location[]
+) {
+    await vscode.commands.executeCommand(
+        "editor.action.peekLocations",
+        uri,
+        position,
+        locations,
+        "peek"
+    );
+}
+
+/**
+ * This function opens at dummy peeked editor in `uri` so that the previously displayed peeked editor gets closed.
+ *
+ * This function exists as a **workaround** for not having any API available to close the previously displayed peeked
+ * editor at `uri`
+ *
+ * **NOTE**: This function call must be immediately followed up by a call to `openPeekedEditorIn` at the same `uri`
```

Comment:
This note indicates to me that this functionality should be moved in to `openPeekedEditorIn` so its impossible to do the wrong thing.